### DISCUSSION TITLE
Alpine.data with initial parameters doesn't work when registered from a bundle

### DISCUSSION THREAD

**Discussion by calvinps** [alpinejs/alpine#1956] [7. Bugs (archived)] [answered]
I am try to create a component via a registered Alpine.data object called "searchSelect" and pass it two initial parameters, but doing so throws the following error:

```
VM1868:3 Uncaught (in promise) ReferenceError: searchSelect is not defined
    at eval (eval at generateFunctionFromString (module.esm.js:1), <anonymous>:3:16)
    at eval (module.esm.js?dd52:1644)
    at tryCatch (module.esm.js?dd52:1668)
    at evaluate (module.esm.js?dd52:1598)
    at eval (module.esm.js?dd52:2809)
    at Function.eval (module.esm.js?dd52:1931)
    at flushHandlers (module.esm.js?dd52:1704)
    at stopDeferring (module.esm.js?dd52:1708)
    at deferHandlingDirectives (module.esm.js?dd52:1711)
    at initTree (module.esm.js?dd52:1887)
```
I registered it like this:

```
import searchSelect from './search-select.js'
Alpine.data('searchSelect', searchSelect)
```
The search-select.js file looks like this:

```
export default (searchCallback = "", selectedCallback = "") => ({
    init() {
        console.log('search select inited');
    },
   ...
})
```
and finally am calling it my html like so:
```
<div x-cloak x-data="searchSelect('searchEndpoint','selectedEndpoint')" >
```

If I remove the parameters and just call it like this `x-data="searchSelect"` then the component does get created (with the initial parameters missing). This leads me to believe there is a bug with creating components when x-data references a function e.g. `x-data="searchSelect()"` instead of a variable e.g. `x-data="searchSelect"`

For reference, I am on alpinejs v3.0.6 and am using alpinejs within a Laravel Livewire project.

**Comment by KevinBatdorf** [comment, marked as answer]
> I am on alpinejs v3.0.6 

What if you try 3.2.3?

**Reply by calvinps** [reply]
Works! Thanks for the suggestion.

**Reply by KevinBatdorf** [reply]
Yeah I think supporting arguments was added sometime between there.

**Comment by TarikAmine** [comment]
I'm having the same problem on 3.14.3, as soon as i add paramters i get `Alpine Expression Error: compo is not defined`

**Reply by ekwoka** [reply]
Need to show code, since this definitely works totally fine under normal circumstances.

**Reply by TarikAmine** [reply]
sorry, i had an other bit of code overriding the registered bundles, should have simplified the code before asking, ty.
